### PR TITLE
Fix quoting of string columns in csv

### PR DESCRIPTION
**Pull request by aeisenberg**
Fixes two issues with CSV exporting and creates a unit test along with a generic framework for creating a mock CLI server. Commit-by-commit review is suggested.

Fixes https://github.com/github/vscode-codeql/issues/1377
Fixes https://github.com/github/vscode-codeql/issues/1376

## Checklist

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [x] Issues have been created for any UI or other user-facing changes made by this pull request.
- [x] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Comment by aeisenberg** [comment]
Adding @edoardopirovano as a reviewer since he is already familiar with this area.

File: extensions/ql-vscode/src/query-history.ts [hunk 1]
```diff
@@ -956,11 +956,11 @@ export class QueryHistoryManager extends DisposableObject {
       void this.tryOpenExternalFile(query.csvPath);
       return;
     }
-    await query.exportCsvResults(this.qs, query.csvPath, () => {
+    if (await query.exportCsvResults(this.qs, query.csvPath)) {
       void this.tryOpenExternalFile(
         query.csvPath
       );
-    });
+    }
```
Comment (edoardopirovano) [inline extensions/ql-vscode/src/query-history.ts:963]: Should we be displaying an error in the `else` case here? I see we `reject()` where we were previously using `showAndLogErrorMessage()` but as far as I can tell we are then swallowing that error.

File: extensions/ql-vscode/src/vscode-tests/no-workspace/run-queries.test.ts [hunk 2]
```diff
@@ -53,6 +58,51 @@ describe('run-queries', () => {
...
 
+  [SELECT_QUERY_NAME, 'other'].forEach(resultSetName => {
+    it(`should export csv results for result set ${resultSetName}`, async () => {
+      const csvLocation = path.join(tmpDir.name, 'test.csv');
+      const qs = createMockQueryServerClient(
+        createMockCliServer({
+          bqrsInfo: [{ 'result-sets': [{ name: resultSetName }, { name: 'hucairz' }] }],
+          bqrsDecode: [{
+            columns: [{ kind: 'NotString' }, { kind: 'String' }],
+            tuples: [['a', 'b'], ['c', 'd']],
```
Comment (edoardopirovano) [inline extensions/ql-vscode/src/vscode-tests/no-workspace/run-queries.test.ts:69]: Let's expand the tuples here to include some quote characters and make sure those get escaped correctly.

File: extensions/ql-vscode/src/run-queries.ts [hunk 3]
```diff
@@ -358,9 +358,12 @@ export class QueryEvaluationInfo {
         pageSize: 100,
         offset: nextOffset,
       });
-      for (const tuple of chunk.tuples) {
-        out.write(tuple.join(',') + '\n');
-      }
+      const quotes = chunk.columns.map(col => col.kind === 'String' ? '"' : '');
+      chunk.tuples.forEach((tuple) => {
+        out.write(tuple.map((v, i) => {
+          return `${quotes[i]}${v}${quotes[i]}`;
```
[the remark this answers is not in the document]
Comment (aeisenberg) [reply extensions/ql-vscode/src/run-queries.ts:378]: Good point. I'll do that.

File: extensions/ql-vscode/src/query-history.ts (hunk 1 above)
Comment (edoardopirovano) [reply extensions/ql-vscode/src/query-history.ts:963]: @aeisenberg I don't think you've addressed this comment. It's less blocking the other others, so I'm happy to approve without any changes here if you don't think any changes are needed, but just checking it wasn't missed accidentally.

Comment (aeisenberg) [reply extensions/ql-vscode/src/query-history.ts:963]: Oh...sorry. I forgot to comment here. The error is not being swallowed. Take a look at `commandRunnerWithProgress`, you can see that there is an error barrier there that catches and logs all thrown and rejected errors. So, you don't ever need to explicitly catch and log an error when you're inside a command as long as that command is wrapped in `commandRunnerWithProgress`.

Comment (edoardopirovano) [reply extensions/ql-vscode/src/query-history.ts:963]: Gotcha, thanks for the explanation 🙂

**Review by edoardopirovano — APPROVED** [review]
LGTM, thanks for addressing my comments!

**Pull request merged**